### PR TITLE
settings: badge app with verify emails and phones

### PR DESCRIPTION
**Pull request by jakob223**
Issue: Y2K-189

CC @keybase/y2ksquad

### CODE REVIEW

File: shared/actions/settings.tsx [hunk 1]
```diff
@@ -393,6 +395,21 @@ const loadSettings = () =>
     })
   })
 
+const computeBadgesOnLoadedSettings = (state, action: SettingsGen.LoadedSettingsPayload) => {
+  const emailCount = (action.payload.emails || I.Map())
+    .map(emailRow => (emailRow.isVerified ? 0 : 1))
```
Comment (jzila) [inline shared/actions/settings.tsx]: I think this would be cleaner as a filter-reduce instead of map-reduce.

Comment (chrisnojima) [reply shared/actions/settings.tsx]: why not just do reduce, you don't need a map or filter first

Comment (jzila) [reply shared/actions/settings.tsx]: Hah yes, good call.

File: shared/actions/settings.tsx [hunk 2]
```diff
@@ -393,6 +395,23 @@ const loadSettings = () =>
...
+    (count, row) => (row.verified ? count : count + 1),
+    0
+  )
+
+  return NotificationsGen.createSetBadgeCounts({
+    counts: I.Map({
+      [Tabs.settingsTab as Tabs.Tab]: emailCount + phoneCount,
+    }) as I.Map<Tabs.Tab, number>,
+  })
+}
```
Comment (jzila) [inline shared/actions/settings.tsx]: Actually one more suggestion to evaluate: can you do this inside `loadSettings` instead? You can `SettingsGen.createLoadedSettings()` and `NotificationsGen.createSetBadgeCounts()` in parallel.

File: shared/actions/settings.tsx [hunk 3]
```diff
@@ -726,6 +729,10 @@ function* settingsSaga(): Saga.SagaGenerator<any, any> {
     SettingsGen.deleteAccountForever,
     deleteAccountForever
   )
+  yield* Saga.chainAction<ConfigGen.BootstrapStatusLoadedPayload>(
```
Comment (buoyad) [inline shared/actions/settings.tsx]: I think it's easier to follow causation if `loadSettings` listens for this action itself

File: shared/actions/settings.tsx (hunk 2 above)
Comment (jakob223) [reply shared/actions/settings.tsx]: How do I return more than one action?

File: shared/actions/settings.tsx (hunk 3 above)
Comment (jakob223) [reply shared/actions/settings.tsx]: hmm, I prefer this way because `loadSettings` would then have to check logged-in-ness in only some cases.

Comment (chrisnojima) [reply shared/actions/settings.tsx]: i dont think this is a safe assumption

Comment (jakob223) [reply shared/actions/settings.tsx]: what is the assumption?

Comment (chrisnojima) [reply shared/actions/settings.tsx]: That this gets called only when your logged in

Comment (jakob223) [reply shared/actions/settings.tsx]: currently the function that is called in this `chainAction` is checking for logged-in-ness before returning the loadSettings action.

Comment (chrisnojima) [reply shared/actions/settings.tsx]: Doesn't mean you won't get logged out. Do your closure relies on being logged in it's safer to validate it yourself vs relying on something upstream

**Comment by jakob223** [comment]
- [ ] after release

**Pull request merged**